server/index.js

```diff
@@ -38,37 +44,59 @@ io.on('connection', socket => {
       socket.emit('full', room);
     } else {
       socket.join(room);
-      rooms[room].push(socket.id.slice(2));
+      rooms[room].push({ peerId: socket.id.slice(2), instrument: undefined });
       console.log('room is', rooms[room]);
+      // update creaorjoin open room table
+      io.emit('give rooms info', getRoomsInfo(rooms));
       // emit message to socket which just joined
       io.to(socket.id).emit('joined', rooms[room]);
       // emit message to other sockets in room
       socket.broadcast.to(room).emit('new peer');
-
+      // socket.emit('give rooms', rooms);
       socket.on('disconnect', () => {
         const socketsInRoom = rooms[room];
         const id = socket.id.slice(2);
-        const index = socketsInRoom.indexOf(id);
-        if (index > -1) {
+        let inRoom = false;
+        let index;
+        // check to make sure peer is in room and get index of peer
+        for (let i = 0; i < socketsInRoom.length; i++) {
```

Comment:
Can import lodash, there are a few instances that would look cleaner with helper functions